master-firmware/src/arms/arm_trajectory_manager.c

```diff
@@ -55,10 +92,18 @@ static THD_FUNCTION(arm_traj_thd, arg)
             error.y = fabsf(target_pos.y - current_pos.y);
             error.z = fabsf(target_pos.z - current_pos.z);
 
+            float error_xy = sqrtf(error.x * error.x + error.y * error.y);
+            bd_manage(&main_arm_traj_manager.blocking_detection_manager_xy, error_xy);
+            bd_manage(&main_arm_traj_manager.blocking_detection_manager_z, error.z);
+
             if (error.x <= main_arm_traj_manager.tol_mm.x &&
                 error.y <= main_arm_traj_manager.tol_mm.y &&
                 error.z <= main_arm_traj_manager.tol_mm.z) {
                 main_arm_traj_manager.state = ARM_READY;
+            } else if (bd_get(&main_arm_traj_manager.blocking_detection_manager_xy)) {
+                main_arm_traj_manager.state = ARM_BLOCKED_XY;
+            } else if (bd_get(&main_arm_traj_manager.blocking_detection_manager_z)) {
+                main_arm_traj_manager.state = ARM_BLOCKED_Z;
```

Comment:
Are the states ARM_BLOCKED_XY and ARM_BLOCKED_Z mutually exclusive?

Reply:
Physically no, but the state variable can only contain one of the events.